agent/api/container.go

```diff
@@ -565,3 +566,23 @@ func (c *Container) BuildContainerDependency(contName string,
 	deps.ContainerDependencies = append(deps.ContainerDependencies, contDep)
 	c.TransitionDependenciesMap[dependentStatus] = deps
 }
+
+// BuildResourceDependency adds a new resource dependency and status that satisfies
+// the dependency
+func (c *Container) BuildResourceDependency(resourceName string,
```

Comment:
The intent is to invoke this in `PostUnmarshalTask()`?

Reply:
Yes, we would add the cgroup/volume dependencies in `PostUnmarshalTask()`.